spec/fastsync/sandbox/fsPCBoundedBlocking.tla

```diff
@@ -0,0 +1,236 @@
+------------------------ MODULE fsPCBoundedBlocking ------------------------
+
+EXTENDS TLC, Integers, Sequences
+CONSTANTS MaxQueueSize
+None == -1  
+
+(*--algorithm message_queue
+variables
+  \* queues prefixed by A, B..so state is easy to inspect
+  Asend_q = <<>>,
+  Breceive_q = <<>>,
+  Cdemux_q = <<>>,
+  msg = [type |-> "none"],
+  peers = {"p1"},
+  done = FALSE, \* set to TRUE when demux is done, send and receive read this and stop
+  
+define
+  FilledQueue(queue) == Len(queue) >= MaxQueueSize
+  FilledQueues == 
+    /\ FilledQueue(Breceive_q)
+    /\ FilledQueue(Asend_q)
+    /\ FilledQueue(Cdemux_q) 
+end define;
+
+macro add_to_blocking_queue(queue, val) begin
+  await Len(queue) < MaxQueueSize;
+  queue := Append(queue, val);
+end macro;
+
+macro read_from_queue(queue) begin
+    await Len(queue) > 0;
+    \* write message in global variable, not possible to return values here or from procedure
+    msg := Head(queue);
+    queue := Tail(queue);
+end macro;
+    
+(* 
+<> bounded blocking queues
+              
+   -------------------------------------
+   |               peer                |
+   |  ..............................   |
+   |  |                            |   |
+   |  v                            |   |         
+   |send                        receive|
+   |  |                            ^   |
+   |  |                            |   |
+   ---+----------------------------+----
+      |                            |
+      |                            |
+   ---+----------------------------+----
+   |  |          this node         |   |
+   |  |                            |   |
+   |  v <> Breceive_q              ^   |
+   |receive                       send |
+   |  |                            ^ <> Asend_q
+   |  |                            |   |
+   |  v <> Cdemux_q (1000)         |   |
+   |  ..............................   |
+   |             demux                 |
+   -------------------------------------
+      
+*)
+
+process demux = "demux"
+variables 
+  counter = 1
+begin Demux:
+  while TRUE do
+    either
+      add_to_blocking_queue(Asend_q, [type |-> "bcStatusRequest", num |-> counter]);
+      counter := counter + 1;
+    or
+      await Len(Cdemux_q) > 0;
+        read_from_queue(Cdemux_q);
+        if msg.type = "bcStatusResponse" then
+          \* terminates on the third status response from peer/ receive
+          if msg.num > 2 /\ msg.num < 1000 then
```

Comment:
Shall we introduce CONSTANTS instead of 2 and 1000?

Reply:
ack